nucypher/acumen/nicknames.py

```diff
@@ -45,18 +44,45 @@ def nicename(symbol):
     return final_word.capitalize()
 
 
-def nickname_from_seed(seed, number_of_pairs=2):
-    # TODO: #1823 - Workaround for new nickname every restart
-    # if not seed:
-    #     raise ValueError("No checksum provided to derive nickname.")
-    symbols = list(symbols_tuple)
-
-    random.seed(seed)
-    pairs = []
-    for pair in range(number_of_pairs):
-        color = random.choice(colors)
-        symbol = random.choice(symbols)
-        symbols.remove(symbol)
-        pairs.append((color, symbol))
-    nickname = " ".join(("{} {}".format(c['color'], nicename(s)) for c, s in pairs))
-    return nickname, pairs
+class NicknameCharacter:
+
+    def __init__(self, symbol, color_name, color_hex):
+        self.symbol = symbol
+        self.color_name = color_name
+        self.color_hex = color_hex
+        self._text = color_name + " " + nicename(symbol)
+
+    def payload(self):
+        return dict(symbol=self.symbol,
+                    color_name=self.color_name,
+                    color_hex=self.color_hex)
+
+    def __str__(self):
+        return self._text
+
+
+class Nickname:
+
+    @classmethod
+    def from_seed(cls, seed, length=2):
+        # TODO: #1823 - Workaround for new nickname every restart
+        # if not seed:
+        #     raise ValueError("No checksum provided to derive nickname.")
+        rng = random.Random(seed)
+        nickname_symbols = rng.sample(symbols, length)
+        nickname_colors = rng.sample(colors, length)
```

Comment:
I don't think there's actually any harm in doing choice with replacement, but currently just keeping the old behavior.

Reply:
Will replacement make cross-language implementations simpler?  If so, let's do it.

Reply:
Replacement here is in the sense that the drawn element is replaced in the list. That is, the symbols/colors can repeat.

Reply:
BTW, this is also a kind of choice we'd better make in this PR. Are repeating symbols ok? I don't see any problem myself, but it wasn't the case in the original code; so maybe there was some reason.

Reply:
I don't recall the reason I opted to prohibit repetition.  There may not be a good one.

I also think that, if we are going to use this, it makes sense to move away from `random.Random(seed)` and to a more cross-language-friendly method of determinism.

Reply:
Good point. Implementing a simple linear congruential generator would work, we don't really need much from randomness here.